Python package

Samsagax opened the pull request:
Create a single python package folder to keep all python sources and use a directory to keep the systemd unit.

When packaging for a distro the python module needs to be built and installed and then the systemd unit should be copied to the appropriate location (typically under `/usr/lib/systemd/system`)

This will need build tools from python:
- python-build
- python-installer
- python-wheel
- python-setuptools

For building and installing the project:
``` shell
$ python -m build --wheel
$ python -m installer dist/*.whl
```

For packagers note that the systemd unit needs to be packaged separately by the distribution packaging system:
```shell
$ python -m build --wheel --no-isolation
$ python -m installer --destdir="$pkgdir" dist/*.whl
$ cp systemd/* /usr/lib/systemd/system/
```

Samsagax committed:
Relocate files in appropriate locations to be packaged

Create a single python package folder to keep all python sources
and use a directory to keep the systemd unit.

When packaging for a distro the python module needs to be built and
installed and then the systemd unit should be copied to the
appropriate location (typically under `/usr/lib/systemd/system`)

Samsagax committed:
Add pyproject.toml file

Add description file for packaging with setuptools. This will need
build tools from python:
- python-build
- python-wheel
- python-setuptools

For building the project:
``` shell
$ python -m build --wheel
$ python -m installer dist/*.whl
```

For packagers:
```shell
$ python -m build --wheel --no-isolation
$ python -m installer --destdir="$pkgdir" dist/*.whl
```

Note that the systemd unit needs to be packaged separately by the
distribution packaging system

Samsagax committed:
Change filenames to snake_case and minor import fix

Add empty __init__.py
Rename files to snake_case
Fix import of internal modules
Fix a function with no return.

pastaq commented:
Seems to be an issue with the .toml

Trackeback:
```
    ~/git/github/samsagax/ryzenadj-controller    python-package    python -m build --wheel                                                                                                      ✔  21s  
* Creating venv isolated environment...
* Installing packages in isolated environment... (setuptools>=61.0)
* Getting build dependencies for wheel...
/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py:108: _BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
  warnings.warn(msg, _BetaConfiguration)
configuration error: `project.scripts.ryzenadj-controller` must be python-entrypoint-reference
DESCRIPTION:
    Reference to a Python object. It is either in the form
    ``importable.module``, or ``importable.module:object.attr``.

GIVEN VALUE:
    "ryzenadj_controller.ryzenadj-controller:main"

OFFENDING RULE: 'format'

DEFINITION:
    {
        "type": "string",
        "format": "python-entrypoint-reference",
        "$comment": "https://packaging.python.org/specifications/entry-points/"
    }
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/pep517/in_process/_in_process.py", line 351, in <module>
    main()
  File "/usr/lib/python3.10/site-packages/pep517/in_process/_in_process.py", line 333, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
  File "/usr/lib/python3.10/site-packages/pep517/in_process/_in_process.py", line 118, in get_requires_for_build_wheel
    return hook(config_settings)
  File "/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/build_meta.py", line 338, in get_requires_for_build_wheel
    return self._get_build_requires(config_settings, requirements=['wheel'])
  File "/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/build_meta.py", line 320, in _get_build_requires
    self.run_setup()
  File "/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/build_meta.py", line 335, in run_setup
    exec(code, locals())
  File "<string>", line 1, in <module>
  File "/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 159, in setup
    dist.parse_config_files()
  File "/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/dist.py", line 867, in parse_config_files
    pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
  File "/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 62, in apply_configuration
    config = read_configuration(filepath, True, ignore_option_errors, dist)
  File "/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 126, in read_configuration
    validate(subset, filepath)
  File "/tmp/build-env-6vqm1ucn/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 51, in validate
    raise ValueError(f"{error}\n{summary}") from None
ValueError: invalid pyproject.toml config: `project.scripts.ryzenadj-controller`.
configuration error: `project.scripts.ryzenadj-controller` must be python-entrypoint-reference

ERROR Backend subprocess exited when trying to invoke get_requires_for_build_wheel

```

pastaq commented on pyproject.toml:
```diff
@@ -0,0 +1,29 @@
+include = ["ryzenadj_controller"]
+
+[project.scripts]
+ryzenadj-controller = "ryzenadj_controller.ryzenadj-controller:main"
```
needs to be `ryzenadj_controller.ryzenadj_controller:main"` sencond _ is - isntead.

pastaq commented:
Alright, got it installed and tried to run it.

```
ryzenadj-controller                                                                                                               ✔ 
Traceback (most recent call last):
  File "/home/blip/.local/bin/ryzenadj-controller", line 5, in <module>
    from ryzenadj_controller.ryzenadj_controller import main
ImportError: cannot import name 'main' from 'ryzenadj_controller.ryzenadj_controller' (/home/blip/.local/lib/python3.10/site-packages/ryzenadj_controller/ryzenadj_controller.py)

```

Samsagax commented:
There is an extra `ryzenadj_controller` in there. There are some changes staged on my side that fix that including a `__main__.py` file to be able to run the module as an executable.

Samsagax committed:
Fix project.script

Samsagax committed:
Add __main__.py to be able to run as a module

- Add __main__.py file and methods to be able to llaunch the module with
`python -m ryzenadj_controller`
- Silence some linting errors and unused imports

Samsagax committed:
Silence some linting errors

Samsagax committed:
Make systemd unit wanted by multi-user target

Samsagax committed:
Fix ryzenadj-controller executable

Samsagax committed:
Move example_client to examples folder

Samsagax committed:
Fix bad refactor name
